ebau_gwr/token_proxy/apps.py

```diff
@@ -1,5 +1,23 @@
 from django.apps import AppConfig
+from django.core.exceptions import ImproperlyConfigured
 
 
 class TokenProxyConfig(AppConfig):
     name = "ebau_gwr.token_proxy"
+
+    def ready(self):
+        from django.conf import settings
+        from .app_settings import DEFAULTS, REQUIRED_SETTINGS
+
+        for setting_key, setting_value in DEFAULTS.items():
+            setattr(
+                settings, setting_key, getattr(settings, setting_key, setting_value),
+            )
+
+        for setting in REQUIRED_SETTINGS:
+            try:
+                getattr(settings, setting)
```

Comment:
You could use `hasattr()` instead to avoid the try/except syntax.

You could even collect the missing settings, so the user knows about all of the missing ones if there's multiples:

```python
missing_settings = [setting in REQUIRED_SETTINGS if not hasattr(settings,setting)]
if missing_settings:
    raise ImproperlyConfigured(f'You need to set "{','.join(missing_settings)}" in your settings module.')
```

Reply:
Nice! Done.